File: MathExt/LESolver.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MathExt
{
	public class LESolver
	{
		/// <summary>
		/// Solves equations that in form Ax=B, where A, B are matrices
		/// </summary>
		public double[] Solve(Matrix A, Matrix B)
		{
			if (B.Rank != 1)
				throw new ArgumentException("Matrix B should have Rank == 1");

			var detA = A.Determinant();

			var addDetA = new double[A.Rank];
			for (int i = 0; i < A.Rank; i++)
			{
				addDetA[i] = A.Replace(i, B).Determinant();
			}

			var result = new double[A.Rank];
			for (int i = 0; i < A.Rank; i++)
			{
				result[i] = addDetA[i] / detA;
			}

			return result;
		}
	}
}

File: cpp/LESolver.cpp
#include "LESolver.h"
#include "Matrix.h"


namespace MathExt
{

	std::vector<double> LESolver::Solve(Matrix *A, Matrix *B)
	{
		if (B->getRank() != 1)
		{
			throw std::invalid_argument("Matrix B should have Rank == 1");
		}

		auto detA = A->Determinant();

		auto addDetA = std::vector<double>(A->getRank());
		for (int i = 0; i < A->getRank(); i++)
		{
//C# TO C++ CONVERTER TODO TASK: There is no direct native C++ equivalent to this .NET String method:
			addDetA[i] = A->Replace(i, B)->Determinant();
		}

		auto result = std::vector<double>(A->getRank());
		for (int i = 0; i < A->getRank(); i++)
		{
			result[i] = addDetA[i] / detA;
		}

		return result;
	}
}
